File: problems/1122-relative-sort-array/solution.py
class Solution:
    def relativeSortArray(self, arr1: List[int], arr2: List[int]) -> List[int]:
        result = []
        remaining = []
        counts = {}
        for num in arr1:
            if num in counts:
                counts[num] += 1
            else:
                counts[num] = 1
        for num in arr2:
            if num in counts:
                result.extend([num] * counts[num])
                del counts[num]
        for num in counts:
            remaining.extend([num] * counts[num])
        remaining.sort()
        return result + remaining

File: problems/1122-relative-sort-array/solution.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int[] count = new int[1001];
        for (int num : arr1) {
            count[num]++;
        }

        int[] result = new int[arr1.length];
        int index = 0;

        for (int num : arr2) {
            while (count[num] > 0) {
                result[index++] = num;
                count[num]--;
            }
        }

        for (int i = 0; i < 1001; i++) {
            while (count[i] > 0) {
                result[index++] = i;
                count[i]--;
            }
        }

        return result;
    }
}